sqlite/migrations/0007_migrate_replications_add_bucket_name.down.sql

```diff
@@ -0,0 +1,43 @@
+-- Adds the "NOT NULL" to `remote_bucket_id` and removes `remote_bucket_name`.
+ALTER TABLE replications RENAME TO _replications_old;
+
+CREATE TABLE replications
+(
+    id                       VARCHAR(16) NOT NULL PRIMARY KEY,
+    org_id                   VARCHAR(16) NOT NULL,
+    name                     TEXT        NOT NULL,
+    description              TEXT,
+    remote_id                VARCHAR(16) NOT NULL,
+    local_bucket_id          VARCHAR(16) NOT NULL,
+    remote_bucket_id         VARCHAR(16) NOT NULL,
+    max_queue_size_bytes     INTEGER     NOT NULL,
+    max_age_seconds          INTEGER     NOT NULL,
+    latest_response_code     INTEGER,
+    latest_error_message     TEXT,
+    drop_non_retryable_data  BOOLEAN     NOT NULL,
+    created_at               TIMESTAMP   NOT NULL,
+    updated_at               TIMESTAMP   NOT NULL,
+
+    CONSTRAINT replications_uniq_orgid_name UNIQUE (org_id, name),
+    FOREIGN KEY (remote_id) REFERENCES remotes (id)
+);
+
+INSERT INTO replications SELECT
+    id,
+    org_id,
+    name,
+    description,
+    remote_id,
+    local_bucket_id,
+    remote_bucket_id,
```

Comment:
How does the 'down' migration work for replications added with a bucket name instead of ID? Won't it fail the constraint check?

Reply:
Hmm that's a good point. It won't fail the NOT NULL because we set it to the dummy `platform.ID(1)` when we create the record based on bucket name. It will however, no longer be a valid replication.

We could delete those rows, but that feels like bad UX/behaviour.

Reply:
I'm not sure there's much else we can do - we don't have anywhere to store the replication bucket name after the down migration, and the old version won't handle it properly.

Reply:
(Much else other than deleting the rows) - would like a third opinion though.

Reply:
Is there a way to look up the bucket ID before downgrading and convert it?

Reply:
After some discussion outside of GitHub, I think we're in agreement that deleting the record is the best option but we should log a message, if possible, for each replication that is deleted.